### PR TITLE
Add a README

### PR DESCRIPTION
**Pull request by olix0r**

### CODE REVIEW

**Comment by olix0r** [comment]
We can improve on this over time -- I just wanted to get something in here to make it more welcoming.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+proxy][linkerd1], it comprises an entirely new codebase implemented in the
+[Rust programming language][rust].
+
+This proxy's features include:
+
+* Transparent, zero-config proxying for HTTP, HTTP/2, and arbitrary TCP protocols.
+* Automatic [Prometheus][prom] metrics export for HTTP and TCP traffic;
+* Transparent, zero-config WebSocket proxying;
+* Opportunistic TLS;
+* [P2C + Peak-EWMA][loadbalancing] HTTP load balancing; and
```
Comment (briansmith) [inline README.md]: I would call out that HTTP load balancing is (1) latency-based, (2) fully-automatic, (3) layer 7 using those keywords. I would also mention that there is also automatic layer 4 load balancing for non-HTTP traffic.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+While the Linkerd2 proxy is heavily influenced by [the Linkerd 1.X
+proxy][linkerd1], it comprises an entirely new codebase implemented in the
+[Rust programming language][rust].
+
+This proxy's features include:
+
+* Transparent, zero-config proxying for HTTP, HTTP/2, and arbitrary TCP protocols.
+* Automatic [Prometheus][prom] metrics export for HTTP and TCP traffic;
+* Transparent, zero-config WebSocket proxying;
+* Opportunistic TLS;
```
Comment (briansmith) [inline README.md]: I would phrase this as "Experimental automatic TLS support (temporarily opportunistic)"

File: README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+* [P2C + Peak-EWMA][loadbalancing] HTTP load balancing; and
+* An on-demand diagnostic `tap` API.
+
+This proxy is primarily intended to run on Linux in containerized
+environments like [Kubernetes][k8s], though it may also work on other
+Unix-like systems (like MacOS).
```
Comment (briansmith) [inline README.md]: s/MacOS/macOS/.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+The proxy supports service discovery via the [`Destination` gRPC
+service][linkerd2-proxy-api] and DNS.
```
Comment (briansmith) [inline README.md]: I would say "The proxy supports service discovery via DNS and via the linkerd2 `Destination` gRPC API" or similar.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+## Code of conduct
+
+This project is for everyone. We ask that our users and contributors take a few
+minutes to review our [code of conduct][coc].
+
+
+## License
+
+Conduit is copyright 2018 the Linkerd authors. All rights reserved.
```
Comment (grampelberg) [inline README.md]: Change to `linkerd2-proxy`?

**Pull request merged**